fastify.js

```diff
@@ -721,9 +722,12 @@ function fastify (options) {
       const reply = new Reply(res, request, childLogger)
 
       if (disableRequestLogging === false) {
-        childLogger.info({ req: request }, 'incoming request')
+        if (createRequestLogMessage) {
+          childLogger.info(createRequestLogMessage(req))
+        } else {
+          childLogger.info({ req: request }, 'incoming request')
+        }
```

Comment:
instead of executing an if condition in the hot path execution of the request, we could just set a default `createRequestLogMessage` function to execute it always

Reply:
Ok I think I know what you mean.  So instead of having:

 if (createRequestLogMessage) {
          childLogger.info(createRequestLogMessage(req))
        } else {
          childLogger.info({ req: request }, 'incoming request')
        }
       
Should I just replace that code with:

createRequestLogMessage(childLogger, req)?

Please let me know if I am incorrect about this.  Thank you.

Reply:
If the goal is just to create the message, what about:
`createRequestLogMessage(req)` and returns a `string` that is used as the log message.

Or we are trying to cover something else there? 🤔

Reply:
@kdrewes I was thinking on:

```js
childLogger.info.call(childLogger, createRequestLogMessage(req))
```

where the default `createRequestLogMessage` returns: `[{ req: request }, 'incoming request']`

Note the returned array and the `call` to avoid to spread the array.

@metcoder95 that usage misses the multiple pino format parameter (the array)